sourcegraph/sourcegraph.proto

```diff
@@ -102,6 +102,10 @@ message Repo {
 	// URL. E.g., "github.com/user/repo".
 	string uri = 1 [(gogoproto.customname) = "URI"];
 
+	// Origin is populated for repos fetched via federation or
+	// discovery. It is the hostname of the host that owns the repo.
+	string origin = 21;
```

Comment:
>  In that case I think origin would be Sourcegraph.com, URI would be golang.org/x/tools, and clone URL would be https://googlesource.com/golang/tools.

In that case, it might be a better comment to say:
```
// Origin is populated for repos fetched via federation 
// discovery. It holds the host from where this repo was
// discovered or federated from.
```
...or something along those lines. When I see "_It is the hostname of the host that owns the repo._" I clearly think of the upstream and based on your description above the upstream is the CloneURL, correct? While the URI is the repo path which may or may not be equal to the clone URL.